src/notifier/keep_alive_monitor.py

```diff
@@ -109,11 +110,14 @@ def _ping_remote(self):
         """
 
         if self._ping_url:
+            param = '-n' if platform.system().lower()=='windows' else '-c'
             logging.debug("Pinging remote keep-alive endpoint")
-            try:
-                urllib.request.urlopen(self._ping_url, timeout=10)
-            except Exception as e:
-                logging.error(f"Failed to ping keep-alive: {e}")
+            command = ['ping', param, '1', self._ping_url]
```

Comment:
What service do you have on the other side checking for those pings?

The publicly available services for health checks usually expose an API GET method that needs to be pinged at an expected interval. I think most people are using this with a service similar to healtchecks.io . And these services aren't compatible with the proposed ping implementation.

Reply:
I was just pinging the machine directly, however, now reading, I get your approach. I might set up an api endpoint.

 Thanks.